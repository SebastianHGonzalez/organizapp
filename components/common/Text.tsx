import { AccessibilityRole, Text as DefaultText } from "react-native";

import Fonts from "@/constants/Fonts";
import { useThemeColors } from "@/hooks/theme/useThemedColors";

export type TextProps = DefaultText["props"] & {
  variant: keyof typeof Fonts;
};

const roleMap: Record<keyof typeof Fonts, AccessibilityRole> = {
  display: "header",
  hero: "header",
  heading1: "header",
  heading2: "header",
  heading3: "header",
  subtitle: "header",
  body: "text",
  bodyBold: "text",
  monospace: "text",
  caption: "text",
  label: "text",
  button: "button",
  input: "text",
  overline: "text",
  mono: "text",
} as const;

export function Text(props: TextProps) {
  const { style, variant, ...otherProps } = props;
  const colors = useThemeColors();

  return (
    <DefaultText
      accessibilityRole={roleMap[variant]}
      style={[
        {
          color: colors.text,
        },
        Fonts[variant],
        style,
      ]}
      {...otherProps}
    />
  );
}
