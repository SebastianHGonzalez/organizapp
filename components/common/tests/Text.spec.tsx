import * as React from "react";
import { render } from "@testing-library/react-native";

import { Text } from "../Text";

it(`renders correctly`, () => {
  const tree = render(<Text variant="body">Snapshot test!</Text>).toJSON();

  expect(tree).toMatchSnapshot();
});
