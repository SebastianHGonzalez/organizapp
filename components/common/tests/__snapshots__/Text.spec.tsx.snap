// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessibilityRole="text"
  style={
    [
      {
        "color": "#18181b",
      },
      {
        "fontFamily": "Poppins",
        "fontSize": 16,
        "fontWeight": "400",
        "letterSpacing": 0.25,
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
